core/node/metadata_calculator/src/helpers.rs

```diff
@@ -552,10 +553,34 @@ impl L1BatchWithLogs {
         };
 
         let load_tree_writes_latency = METRICS.start_load_stage(LoadChangesStage::LoadTreeWrites);
-        let tree_writes = storage
-            .blocks_dal()
-            .get_tree_writes(l1_batch_number)
-            .await?;
+
+        let tree_writes = if wait_for_tree_writes {
+            // If `wait_for_tree_writes` flag is set then we expect that waiting for tree writes
```

Comment:
Bikeshedding: This looks awkward; I'd structure it as a timeout with a polling loop inside:

```rust
let tree_writes = tokio::time::timeout(_, async {
    loop {
        if let Some(tree_writes) = storage
            .blocks_dal()
            .get_tree_writes(l1_batch_number)
            .await?
        {
            break anyhow::Ok(tree_writes); 
        }
        tokio::time::sleep(_).await;
    }
})
.await
.ok()
.transpose()?;
```

... and maybe extract it to a method.